relayer/submit_transaction.go

```diff
@@ -119,3 +137,68 @@ func checkForMinClaimBalance(ctx context.Context, ec extethclient.EthClient) (*b
 
 	return gasPrice, nil
 }
+
+// simulateExecute calls the forwarder's execute method (defined in Forwarder.sol)
+// with CallContract which executes the method call without mining it into the blockchain.
+// https://pkg.go.dev/github.com/ethereum/go-ethereum/ethclient#Client.CallContract
+func simulateExecute(
+	ctx context.Context,
+	ec extethclient.EthClient,
+	reqForwarderAddr *ethcommon.Address,
+	txOpts *bind.TransactOpts,
+	forwarderReq gsnforwarder.IForwarderForwardRequest,
+	domainSeparator [32]byte,
+	sig []byte,
+) ([]byte, error) {
```

Comment:
```suggestion
) error {
```

I'd get rid of this first return value since it's never used